### ISSUE TITLE
directory listing fails on the website 

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxelib#281]
e.g. http://lib.haxe.org/p/hxnodejs/4.0.3/files/src

**Comment by ibilon** [comment]
Looks like it's only for newly uploaded versions http://lib.haxe.org/p/hxnodejs/4.0.2/files/

**Comment by nadako** [comment]
Nah, it's the same http://lib.haxe.org/p/hxnodejs/4.0.2/files/src/

**Comment by ibilon** [comment]
Hmmm http://lib.haxe.org/p/lime/2.9.0/files/lime/tools/mpeg/audio/ this works

**Comment by ibilon** [comment]
This doesn't work http://lib.haxe.org/p/haxelib_client/3.2.0-rc.3/files/src/

So I guess `src` folders aren't working.

**Comment by nadako** [comment]
that's really weird

**Comment by ibilon** [comment]
Seems random http://lib.haxe.org/p/openfl-samples/3.1.0/files/openfl-samples/ http://lib.haxe.org/p/openfl-samples/3.3.1/files/openfl-samples/
maybe some decompression/file creation issue?

**Comment by nadako** [comment]
Hmm, I think it's deployed, but the problem is still there. I wonder why it worked for me locally.

**Comment by nadako** [comment]
It works for new uploads (http://lib.haxe.org/p/hxnodejs/4.0.4/files/src), so this is probably a cache issue.

**Comment by ibilon** [comment]
That's still a strange bug even if the cache is behaving weirdly

That's only on directory
http://lib.haxe.org/p/openfl-samples/3.3.1/files/openfl-samples/
http://lib.haxe.org/p/openfl-samples/3.3.1/files/openfl-samples/README.md
you can access the files if you know about them
and not all directories of the same lib version fail
http://lib.haxe.org/p/openfl-samples/3.3.1/files/openfl-samples/demos/NyanCat/

**Comment by nadako** [comment]
@andyli or @jasononeil (still not sure who to ping) this might require some server maintainance: we need to clear zip caches.

I see this ufront task that should do that, but it requires project/version, and I think we need to do that for ALL projects and versions: https://github.com/HaxeFoundation/haxelib/blob/development/src/website/tasks/HaxelibCacheTasks.hx#L21

**Comment by nadako** [comment]
@ibilon I _think_ I fixed that, see e7e6b615f937c3dc888bf2bf0164288ad8444b8e

**Comment by nadako** [comment]
I.e. that depends on whether an entry was added to the zip file for a directory or not.

**Comment by ibilon** [comment]
Oh I see, make sense.

**Comment by ibilon** [comment]
> I see this ufront task that should do that, but it requires project/version, and I think we need to do that for ALL projects and versions

There's always `clearAll` https://github.com/HaxeFoundation/haxelib/blob/development/src/website/tasks/HaxelibCacheTasks.hx#L15 it's a bit overkill but at least it'd work.

**Comment by nadako** [comment]
I was toying with `tasks.n` today and couldn't get it to work - it seems that there's something wrong with minject or injector configuration, so `HaxelibCacheTasks.api` doesn't get set to a `DBCacheApi` instance and it crashes.

Could someone with ufront/minject knowledge help me here?

**Comment by ibilon** [comment]
I looked at this, it does the injection but with the value `null`.
most likely since `injector.getInstance(DBCacheApi)` throws `Uncaught exception - Error while getting mapping response: No mapping defined for class "ufront.cache.DBCacheApi" name "null"`

But the class is correctly returned in this https://github.com/HaxeFoundation/haxelib/blob/development/src/website/Tasks.hx#L34

Still fails even if I add `tasks.injector.map( DBCacheApi ).toClass( DBCacheApi );`

Well anyway, I get the use of injection, but here it's a 100 lines software that's not meant to be used as a library where we do `obj.fn()`,
removing the injection and doing `new DBCacheApi().clearAll(); ` would be so much simpler.

**Comment by markknol** [comment]
Any news on how we could resolve this?
